segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ConcatOperation.java

```diff
@@ -179,8 +179,9 @@ private void logEnd() {
                                 targetSegmentMetadata.setChunkCount(targetSegmentMetadata.getChunkCount() + sourceSegmentMetadata.getChunkCount());
 
                                 // Delete read index block entries for source.
-                                chunkedSegmentStorage.deleteBlockIndexEntriesForChunk(txn, sourceSegment, sourceSegmentMetadata.getStartOffset(), sourceSegmentMetadata.getLength());
-
+                                if (!sourceSegmentMetadata.isStorageSystemSegment()) {
```

Comment:
It would be nice to explain (In a comment) why we are not doing this for storage system segments.